File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-com-5-4/CoalDelivery.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class CoalDelivery : ISerialized
    {
        public CoalDelivery(XmlNode xmlNode)
        {
            XmlNode deliveryPointNode = xmlNode.SelectSingleNode("deliveryPoint");
            
            if (deliveryPointNode != null)
            {
                if (deliveryPointNode.Attributes["href"] != null || deliveryPointNode.Attributes["id"] != null) 
                {
                    if (deliveryPointNode.Attributes["id"] != null) 
                    {
                        deliveryPointIDRef_ = deliveryPointNode.Attributes["id"].Value;
                        CoalDeliveryPoint ob = new CoalDeliveryPoint(deliveryPointNode);
                        IDManager.SetID(deliveryPointIDRef_, ob);
                    }
                    else if (deliveryPointNode.Attributes["href"] != null)
                    {
                        deliveryPointIDRef_ = deliveryPointNode.Attributes["href"].Value;
                    }
                    else
                    {
                        deliveryPoint_ = new CoalDeliveryPoint(deliveryPointNode);
                    }
                }
                else
                {
                    deliveryPoint_ = new CoalDeliveryPoint(deliveryPointNode);
                }
            }
            
        
            XmlNode deliveryAtSourceNode = xmlNode.SelectSingleNode("deliveryAtSource");
            
            if (deliveryAtSourceNode != null)
            {
                if (deliveryAtSourceNode.Attributes["href"] != null || deliveryAtSourceNode.Attributes["id"] != null) 
                {
                    if (deliveryAtSourceNode.Attributes["id"] != null) 
                    {
                        deliveryAtSourceIDRef_ = deliveryAtSourceNode.Attributes["id"].Value;
                        XsdTypeBoolean ob = new XsdTypeBoolean(deliveryAtSourceNode);
                        IDManager.SetID(deliveryAtSourceIDRef_, ob);
                    }
                    else if (deliveryAtSourceNode.Attributes["href"] != null)
                    {
                        deliveryAtSourceIDRef_ = deliveryAtSourceNode.Attributes["href"].Value;
                    }
                    else
                    {
                        deliveryAtSource_ = new XsdTypeBoolean(deliveryAtSourceNode);
                    }
                }
                else
                {
                    deliveryAtSource_ = new XsdTypeBoolean(deliveryAtSourceNode);
                }
            }
            
        
            XmlNode quantityVariationAdjustmentNode = xmlNode.SelectSingleNode("quantityVariationAdjustment");
            
            if (quantityVariationAdjustmentNode != null)
            {
                if (quantityVariationAdjustmentNode.Attributes["href"] != null || quantityVariationAdjustmentNode.Attributes["id"] != null) 
                {
                    if (quantityVariationAdjustmentNode.Attributes["id"] != null) 
                    {
                        quantityVariationAdjustmentIDRef_ = quantityVariationAdjustmentNode.Attributes["id"].Value;
                        XsdTypeBoolean ob = new XsdTypeBoolean(quantityVariationAdjustmentNode);
                        IDManager.SetID(quantityVariationAdjustmentIDRef_, ob);
                    }
                    else if (quantityVariationAdjustmentNode.Attributes["href"] != null)
                    {
                        quantityVariationAdjustmentIDRef_ = quantityVariationAdjustmentNode.Attributes["href"].Value;
                    }
                    else
                    {
                        quantityVariationAdjustment_ = new XsdTypeBoolean(quantityVariationAdjustmentNode);
                    }
                }
                else
                {
                    quantityVariationAdjustment_ = new XsdTypeBoolean(quantityVariationAdjustmentNode);
                }
            }
            
        
            XmlNode transportationEquipmentNode = xmlNode.SelectSingleNode("transportationEquipment");
            
            if (transportationEquipmentNode != null)
            {
                if (transportationEquipmentNode.Attributes["href"] != null || transportationEquipmentNode.Attributes["id"] != null) 
                {
                    if (transportationEquipmentNode.Attributes["id"] != null) 
                    {
                        transportationEquipmentIDRef_ = transportationEquipmentNode.Attributes["id"].Value;
                        CoalTransportationEquipment ob = new CoalTransportationEquipment(transportationEquipmentNode);
                        IDManager.SetID(transportationEquipmentIDRef_, ob);
                    }
                    else if (transportationEquipmentNode.Attributes["href"] != null)
                    {
                        transportationEquipmentIDRef_ = transportationEquipmentNode.Attributes["href"].Value;
                    }
                    else
                    {
                        transportationEquipment_ = new CoalTransportationEquipment(transportationEquipmentNode);
                    }
                }
                else
                {
                    transportationEquipment_ = new CoalTransportationEquipment(transportationEquipmentNode);
                }
            }
            
        
            XmlNode riskNode = xmlNode.SelectSingleNode("risk");
            
            if (riskNode != null)
            {
                if (riskNode.Attributes["href"] != null || riskNode.Attributes["id"] != null) 
                {
                    if (riskNode.Attributes["id"] != null) 
                    {
                        riskIDRef_ = riskNode.Attributes["id"].Value;
                        CommodityDeliveryRisk ob = new CommodityDeliveryRisk(riskNode);
                        IDManager.SetID(riskIDRef_, ob);
                    }
                    else if (riskNode.Attributes["href"] != null)
                    {
                        riskIDRef_ = riskNode.Attributes["href"].Value;
                    }
                    else
                    {
                        risk_ = new CommodityDeliveryRisk(riskNode);
                    }
                }
                else
                {
                    risk_ = new CommodityDeliveryRisk(riskNode);
                }
            }
            
        
        }
        
    
        #region DeliveryPoint_
        private CoalDeliveryPoint deliveryPoint_;
        public CoalDeliveryPoint DeliveryPoint_
        {
            get
            {
                if (this.deliveryPoint_ != null)
                {
                    return this.deliveryPoint_; 
                }
                else if (this.deliveryPointIDRef_ != null)
                {
                    deliveryPoint_ = IDManager.getID(deliveryPointIDRef_) as CoalDeliveryPoint;
                    return this.deliveryPoint_; 
                }
                else
                {
                      return this.deliveryPoint_; 
                }
            }
            set
            {
                if (this.deliveryPoint_ != value)
                {
                    this.deliveryPoint_ = value;
                }
            }
        }
        #endregion
        
        public string deliveryPointIDRef_ { get; set; }
        #region DeliveryAtSource_
        private XsdTypeBoolean deliveryAtSource_;
        public XsdTypeBoolean DeliveryAtSource_
        {
            get
            {
                if (this.deliveryAtSource_ != null)
                {
                    return this.deliveryAtSource_; 
                }
                else if (this.deliveryAtSourceIDRef_ != null)
                {
                    deliveryAtSource_ = IDManager.getID(deliveryAtSourceIDRef_) as XsdTypeBoolean;
                    return this.deliveryAtSource_; 
                }
                else
                {
                      return this.deliveryAtSource_; 
                }
            }
            set
            {
                if (this.deliveryAtSource_ != value)
                {
                    this.deliveryAtSource_ = value;
                }
            }
        }
        #endregion
        
        public string deliveryAtSourceIDRef_ { get; set; }
        #region QuantityVariationAdjustment_
        private XsdTypeBoolean quantityVariationAdjustment_;
        public XsdTypeBoolean QuantityVariationAdjustment_
        {
            get
            {
                if (this.quantityVariationAdjustment_ != null)
                {
                    return this.quantityVariationAdjustment_; 
                }
                else if (this.quantityVariationAdjustmentIDRef_ != null)
                {
                    quantityVariationAdjustment_ = IDManager.getID(quantityVariationAdjustmentIDRef_) as XsdTypeBoolean;
                    return this.quantityVariationAdjustment_; 
                }
                else
                {
                      return this.quantityVariationAdjustment_; 
                }
            }
            set
            {
                if (this.quantityVariationAdjustment_ != value)
                {
                    this.quantityVariationAdjustment_ = value;
                }
            }
        }
        #endregion
        
        public string quantityVariationAdjustmentIDRef_ { get; set; }
        #region TransportationEquipment_
        private CoalTransportationEquipment transportationEquipment_;
        public CoalTransportationEquipment TransportationEquipment_
        {
            get
            {
                if (this.transportationEquipment_ != null)
                {
                    return this.transportationEquipment_; 
                }
                else if (this.transportationEquipmentIDRef_ != null)
                {
                    transportationEquipment_ = IDManager.getID(transportationEquipmentIDRef_) as CoalTransportationEquipment;
                    return this.transportationEquipment_; 
                }
                else
                {
                      return this.transportationEquipment_; 
                }
            }
            set
            {
                if (this.transportationEquipment_ != value)
                {
                    this.transportationEquipment_ = value;
                }
            }
        }
        #endregion
        
        public string transportationEquipmentIDRef_ { get; set; }
        #region Risk_
        private CommodityDeliveryRisk risk_;
        public CommodityDeliveryRisk Risk_
        {
            get
            {
                if (this.risk_ != null)
                {
                    return this.risk_; 
                }
                else if (this.riskIDRef_ != null)
                {
                    risk_ = IDManager.getID(riskIDRef_) as CommodityDeliveryRisk;
                    return this.risk_; 
                }
                else
                {
                      return this.risk_; 
                }
            }
            set
            {
                if (this.risk_ != value)
                {
                    this.risk_ = value;
                }
            }
        }
        #endregion
        
        public string riskIDRef_ { get; set; }
        public string choiceStr_0;
        
    
        
    
    }
    
}

File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-com-5-4/CoalDelivery.cpp
// CoalDelivery.cpp 
#include "CoalDelivery.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

CoalDelivery::CoalDelivery(TiXmlNode* xmlNode)
: ISerialized(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //deliveryPointNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* deliveryPointNode = xmlNode->FirstChildElement("deliveryPoint");

   if(deliveryPointNode){deliveryPointIsNull_ = false;}
   else{deliveryPointIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- deliveryPointNode , address : " << deliveryPointNode << std::endl;
   #endif
   if(deliveryPointNode)
   {
      if(deliveryPointNode->Attribute("href") || deliveryPointNode->Attribute("id"))
      {
          if(deliveryPointNode->Attribute("id"))
          {
             deliveryPointIDRef_ = deliveryPointNode->Attribute("id");
             deliveryPoint_ = boost::shared_ptr<CoalDeliveryPoint>(new CoalDeliveryPoint(deliveryPointNode));
             deliveryPoint_->setID(deliveryPointIDRef_);
             IDManager::instance().setID(deliveryPointIDRef_,deliveryPoint_);
          }
          else if(deliveryPointNode->Attribute("href")) { deliveryPointIDRef_ = deliveryPointNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { deliveryPoint_ = boost::shared_ptr<CoalDeliveryPoint>(new CoalDeliveryPoint(deliveryPointNode));}
   }

   //deliveryAtSourceNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* deliveryAtSourceNode = xmlNode->FirstChildElement("deliveryAtSource");

   if(deliveryAtSourceNode){deliveryAtSourceIsNull_ = false;}
   else{deliveryAtSourceIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- deliveryAtSourceNode , address : " << deliveryAtSourceNode << std::endl;
   #endif
   if(deliveryAtSourceNode)
   {
      if(deliveryAtSourceNode->Attribute("href") || deliveryAtSourceNode->Attribute("id"))
      {
          if(deliveryAtSourceNode->Attribute("id"))
          {
             deliveryAtSourceIDRef_ = deliveryAtSourceNode->Attribute("id");
             deliveryAtSource_ = boost::shared_ptr<XsdTypeBoolean>(new XsdTypeBoolean(deliveryAtSourceNode));
             deliveryAtSource_->setID(deliveryAtSourceIDRef_);
             IDManager::instance().setID(deliveryAtSourceIDRef_,deliveryAtSource_);
          }
          else if(deliveryAtSourceNode->Attribute("href")) { deliveryAtSourceIDRef_ = deliveryAtSourceNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { deliveryAtSource_ = boost::shared_ptr<XsdTypeBoolean>(new XsdTypeBoolean(deliveryAtSourceNode));}
   }

   //quantityVariationAdjustmentNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* quantityVariationAdjustmentNode = xmlNode->FirstChildElement("quantityVariationAdjustment");

   if(quantityVariationAdjustmentNode){quantityVariationAdjustmentIsNull_ = false;}
   else{quantityVariationAdjustmentIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- quantityVariationAdjustmentNode , address : " << quantityVariationAdjustmentNode << std::endl;
   #endif
   if(quantityVariationAdjustmentNode)
   {
      if(quantityVariationAdjustmentNode->Attribute("href") || quantityVariationAdjustmentNode->Attribute("id"))
      {
          if(quantityVariationAdjustmentNode->Attribute("id"))
          {
             quantityVariationAdjustmentIDRef_ = quantityVariationAdjustmentNode->Attribute("id");
             quantityVariationAdjustment_ = boost::shared_ptr<XsdTypeBoolean>(new XsdTypeBoolean(quantityVariationAdjustmentNode));
             quantityVariationAdjustment_->setID(quantityVariationAdjustmentIDRef_);
             IDManager::instance().setID(quantityVariationAdjustmentIDRef_,quantityVariationAdjustment_);
          }
          else if(quantityVariationAdjustmentNode->Attribute("href")) { quantityVariationAdjustmentIDRef_ = quantityVariationAdjustmentNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { quantityVariationAdjustment_ = boost::shared_ptr<XsdTypeBoolean>(new XsdTypeBoolean(quantityVariationAdjustmentNode));}
   }

   //transportationEquipmentNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* transportationEquipmentNode = xmlNode->FirstChildElement("transportationEquipment");

   if(transportationEquipmentNode){transportationEquipmentIsNull_ = false;}
   else{transportationEquipmentIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- transportationEquipmentNode , address : " << transportationEquipmentNode << std::endl;
   #endif
   if(transportationEquipmentNode)
   {
      if(transportationEquipmentNode->Attribute("href") || transportationEquipmentNode->Attribute("id"))
      {
          if(transportationEquipmentNode->Attribute("id"))
          {
             transportationEquipmentIDRef_ = transportationEquipmentNode->Attribute("id");
             transportationEquipment_ = boost::shared_ptr<CoalTransportationEquipment>(new CoalTransportationEquipment(transportationEquipmentNode));
             transportationEquipment_->setID(transportationEquipmentIDRef_);
             IDManager::instance().setID(transportationEquipmentIDRef_,transportationEquipment_);
          }
          else if(transportationEquipmentNode->Attribute("href")) { transportationEquipmentIDRef_ = transportationEquipmentNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { transportationEquipment_ = boost::shared_ptr<CoalTransportationEquipment>(new CoalTransportationEquipment(transportationEquipmentNode));}
   }

   //riskNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* riskNode = xmlNode->FirstChildElement("risk");

   if(riskNode){riskIsNull_ = false;}
   else{riskIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- riskNode , address : " << riskNode << std::endl;
   #endif
   if(riskNode)
   {
      if(riskNode->Attribute("href") || riskNode->Attribute("id"))
      {
          if(riskNode->Attribute("id"))
          {
             riskIDRef_ = riskNode->Attribute("id");
             risk_ = boost::shared_ptr<CommodityDeliveryRisk>(new CommodityDeliveryRisk(riskNode));
             risk_->setID(riskIDRef_);
             IDManager::instance().setID(riskIDRef_,risk_);
          }
          else if(riskNode->Attribute("href")) { riskIDRef_ = riskNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { risk_ = boost::shared_ptr<CommodityDeliveryRisk>(new CommodityDeliveryRisk(riskNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<CoalDeliveryPoint> CoalDelivery::getDeliveryPoint()
{
   if(!this->deliveryPointIsNull_){
        if(deliveryPointIDRef_ != ""){
             return boost::shared_static_cast<CoalDeliveryPoint>(IDManager::instance().getID(deliveryPointIDRef_));
        }else{
             return this->deliveryPoint_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<CoalDeliveryPoint>();
   }
}
boost::shared_ptr<XsdTypeBoolean> CoalDelivery::getDeliveryAtSource()
{
   if(!this->deliveryAtSourceIsNull_){
        if(deliveryAtSourceIDRef_ != ""){
             return boost::shared_static_cast<XsdTypeBoolean>(IDManager::instance().getID(deliveryAtSourceIDRef_));
        }else{
             return this->deliveryAtSource_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeBoolean>();
   }
}
boost::shared_ptr<XsdTypeBoolean> CoalDelivery::getQuantityVariationAdjustment()
{
   if(!this->quantityVariationAdjustmentIsNull_){
        if(quantityVariationAdjustmentIDRef_ != ""){
             return boost::shared_static_cast<XsdTypeBoolean>(IDManager::instance().getID(quantityVariationAdjustmentIDRef_));
        }else{
             return this->quantityVariationAdjustment_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeBoolean>();
   }
}
boost::shared_ptr<CoalTransportationEquipment> CoalDelivery::getTransportationEquipment()
{
   if(!this->transportationEquipmentIsNull_){
        if(transportationEquipmentIDRef_ != ""){
             return boost::shared_static_cast<CoalTransportationEquipment>(IDManager::instance().getID(transportationEquipmentIDRef_));
        }else{
             return this->transportationEquipment_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<CoalTransportationEquipment>();
   }
}
boost::shared_ptr<CommodityDeliveryRisk> CoalDelivery::getRisk()
{
   if(!this->riskIsNull_){
        if(riskIDRef_ != ""){
             return boost::shared_static_cast<CommodityDeliveryRisk>(IDManager::instance().getID(riskIDRef_));
        }else{
             return this->risk_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<CommodityDeliveryRisk>();
   }
}
}

